service/api_task.go

```diff
@@ -584,6 +569,22 @@ func assignNextAvailableTask(ctx context.Context, taskQueue *model.TaskQueue, di
 				if numHosts > nextTask.TaskGroupMaxHosts {
 					dispatchRace = fmt.Sprintf("tasks found on %d hosts", numHosts)
 				}
+
+				// if the previous task in the group has yet to run and should run, then wait for it
+				if nextTask.TaskGroupOrder != 0 && nextTask.TaskGroupMaxHosts == 1 {
```

Comment:
should this be an else if in case there was already a race caught by the numHosts > nextTask.TaskGroupMaxHosts check?

Reply:
Yes, good point